packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,860 @@
+        end_epoch_exclusive: intent.header.endEpochExclusive,
+        notary_public_key: RadixSpecificClient.retPublicKeyToGatewayPublicKey(intent.header.notaryPublicKey),
+        notary_is_signatory: intent.header.notaryIsSignatory,
+        tip_percentage: intent.header.tipPercentage,
+        nonce: intent.header.nonce,
+        signer_public_keys: [RadixSpecificClient.retPublicKeyToGatewayPublicKey(intent.header.notaryPublicKey)],
+        // TODO: Add message
+        flags: {
+          assume_all_signature_proofs: false,
+          skip_epoch_check: false,
+          use_free_credit: false,
+        },
+      },
+    }
+
+    return this.innerGatewayClient.transaction.innerClient.transactionPreview(request)
+  }
+
+  private static retPublicKeyToGatewayPublicKey(publicKey: PublicKey): GatewayPublicKey {
+    switch (publicKey.curve) {
+      case 'Secp256k1':
+        return {
+          key_type: 'EcdsaSecp256k1',
+          key_hex: publicKey.hex(),
+        }
+      case 'Ed25519':
+        return {
+          key_type: 'EddsaEd25519',
+          key_hex: publicKey.hex(),
+        }
+    }
+  }
+  // #endregion Private Methods
+}
+
+/**
+ * Custom Radix client
+ */
+
+export default class Client extends BaseXChainClient {
+  private radixSpecificClient: RadixSpecificClient
+  private curve: Curve
+  constructor(
+    {
+      network = Network.Mainnet,
+      phrase,
+      rootDerivationPaths = xrdRootDerivationPaths,
+      feeBounds = {
+        lower: 0,
+        upper: 3,
+      },
+    }: XChainClientParams,
+    curve: Curve,
+  ) {
+    super(RadixChain, {
+      network: network,
+      phrase: phrase,
+      rootDerivationPaths: rootDerivationPaths,
+      feeBounds: feeBounds,
+    })
+    this.curve = curve
+    this.radixSpecificClient = new RadixSpecificClient(xChainJsNetworkToRadixNetworkId(network))
+  }
+
+  public get radixClient(): RadixSpecificClient {
+    return this.radixSpecificClient
+  }
+
+  setNetwork(network: Network): void {
+    super.setNetwork(network)
+    this.radixSpecificClient.networkId = xChainJsNetworkToRadixNetworkId(network)
+  }
+
+  /**
+   * Get an estimated fee for a test transaction that involves sending
+   * XRD from one account to another
+   *
+   * @returns {Fee} An estimated fee
+   */
+  async getFees(): Promise<Fees> {
+    // TODO: This can fail if we use it on stokenet, we need to replace these with network aware
+    // addresses.
+    const feesInXrd = await this.radixSpecificClient
+      .constructSimpleTransferIntent(
+        feesEstimationPublicKeys[this.getRadixNetwork()].from,
+        feesEstimationPublicKeys[this.getRadixNetwork()].to,
+        feesEstimationPublicKeys[this.getRadixNetwork()].resourceAddress,
+        1,
+        new PublicKey.Ed25519(
+          Convert.HexString.toUint8Array(feesEstimationPublicKeys[this.getRadixNetwork()].publicKey),
+        ),
+        '',
+      )
+      .then((result) => result.fees)
```

Comment:
As we saw yesterday we need some padding since the preview doesn't account for everything.

Reply:
added

Reply:
done